corehq/apps/locations/dbaccessors.py

```diff
@@ -92,3 +92,45 @@ def user_ids_at_accessible_locations(domain_name, user):
     from corehq.apps.locations.models import SQLLocation
     accessible_location_ids = SQLLocation.active_objects.accessible_location_ids(domain_name, user)
     return user_ids_at_locations(accessible_location_ids)
+
+
+def assigned_location_by_user_id(domain, location_ids):
+    """
+    Returns {user_id: [location_id, location_id, ...], ...}
+    """
+    result = (
+        UserES()
+        .domain(domain)
+        .location(location_ids)
+        .non_null('assigned_location_ids')
+        .fields(['assigned_location_ids', '_id'])
+        .run().hits
+    )
+    ret = {}
+    for r in result:
+        if 'assigned_location_ids' in r:
+            locs = r['assigned_location_ids']
+            if not isinstance(locs, list):
+                locs = [r['assigned_location_ids']]
+            ret[r['_id']] = locs
+    return ret
+
+
+def primary_location_by_user_id(domain, location_ids):
```

Comment:
i was a little confused by the name of this function because it seems like it should take a user_id as a parameter. could we rename it to something like `get_user_ids_from_primary_location_ids`? same with the one above it too